server/src/main/java/org/candlepin/controller/Entitler.java

```diff
@@ -339,8 +313,45 @@ private DeveloperProducts getDevProductMap(Consumer consumer, String sku) {
         for (ConsumerInstalledProduct ip : consumer.getInstalledProducts()) {
             devProductIds.add(ip.getProductId());
         }
-        List<Product> prods = productAdapter.getProductsByIds(consumer.getOwner(), devProductIds);
-        return new DeveloperProducts(sku, prods);
+
+        Owner owner = consumer.getOwner();
+        Set<Product> products = new HashSet<Product>();
+
+        for (Product product : productAdapter.getProductsByIds(consumer.getOwner(), devProductIds)) {
+            for (ProductContent pc : product.getProductContent()) {
+                pc.getContent().addOwner(owner);
+
+                // VERSIONING TODO: We may need to hit the content curator here. Hopefully the update on
```

Comment:
Potentially. Currently all the unit and spec tests seem to be okay with it, but I want to leave it in until we give it a bit more hands-on testing via IT.